ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/data/processed/VariableProcessor.kt

```diff
@@ -31,21 +31,46 @@ internal object VariableProcessor {
         rcPackage: Package,
         locale: Locale,
     ): String? {
-        return when (variableName) {
-            "app_name" -> variableDataProvider.applicationName
-            "price" -> variableDataProvider.localizedPrice(rcPackage)
-            "price_per_period" -> variableDataProvider.localizedPricePerPeriod(rcPackage)
-            "total_price_and_per_month" -> variableDataProvider.localizedPriceAndPerMonth(rcPackage)
-            "product_name" -> variableDataProvider.productName(rcPackage)
-            "sub_period" -> variableDataProvider.periodName(rcPackage)
-            "sub_price_per_month" -> variableDataProvider.localizedPricePerMonth(rcPackage, locale)
-            "sub_duration" -> variableDataProvider.subscriptionDuration(rcPackage)
-            "sub_offer_duration" -> variableDataProvider.introductoryOfferDuration(rcPackage)
-            "sub_offer_price" -> variableDataProvider.localizedIntroductoryOfferPrice(rcPackage)
+        return when (VariableName.valueOfIdentifier(variableName)) {
+            VariableName.APP_NAME -> variableDataProvider.applicationName
+            VariableName.PRICE -> variableDataProvider.localizedPrice(rcPackage)
+            VariableName.PRICE_PER_PERIOD -> variableDataProvider.localizedPricePerPeriod(rcPackage)
+            VariableName.TOTAL_PRICE_AND_PER_MONTH -> variableDataProvider.localizedPriceAndPerMonth(
+                rcPackage,
+            )
+            VariableName.PRODUCT_NAME -> variableDataProvider.productName(rcPackage)
+            VariableName.SUB_PERIOD -> variableDataProvider.periodName(rcPackage)
+            VariableName.SUB_PRICE_PER_MONTH -> variableDataProvider.localizedPricePerMonth(
+                rcPackage,
+                locale,
+            )
+            VariableName.SUB_DURATION -> variableDataProvider.subscriptionDuration(rcPackage)
+            VariableName.SUB_OFFER_DURATION -> variableDataProvider.introductoryOfferDuration(rcPackage)
+            VariableName.SUB_OFFER_PRICE -> variableDataProvider.localizedIntroductoryOfferPrice(rcPackage)
             else -> {
                 Logger.e("Unknown variable: $variableName")
                 null
             }
         }
     }
+
+    private enum class VariableName(val identifier: String) {
+        APP_NAME("app_name"),
+        PRICE("price"),
+        PRICE_PER_PERIOD("price_per_period"),
+        TOTAL_PRICE_AND_PER_MONTH("total_price_and_per_month"),
+        PRODUCT_NAME("product_name"),
+        SUB_PERIOD("sub_period"),
+        SUB_PRICE_PER_MONTH("sub_price_per_month"),
+        SUB_DURATION("sub_duration"),
+        SUB_OFFER_DURATION("sub_offer_duration"),
+        SUB_OFFER_PRICE("sub_offer_price"),
+        ;
+        
+        companion object {
+            fun valueOfIdentifier(identifier: String): VariableName? {
+                return values().find { it.identifier == identifier }
```

Comment:
Sorry I think I commented on the commit directly.

Is there a reason this has to be linear?
In Swift I would build a `Map<String, VariableName>` to avoid this, seems like an easy win?

Reply:
how would you ensure the map is updated if new values are added to the enum?

Reply:
I guess we could build the map with the `values` method once. Small win I guess.

Reply:
> how would you ensure the map is updated if new values are added to the enum?

By constructing it automatically:
https://github.com/RevenueCat/purchases-ios/blob/main/Sources/Purchasing/PackageType.swift#L105-L108

Reply:
I think it's worth it. If you have maybe on average 6 variables in your strings, that's iterating the list of variables 6 times for no reason.

Reply:
+1

Reply:
Sure. I didn't think initially to do it with an associateBy and build the map dynamically. It makes sense!